server/src/main/java/io/crate/fdw/JdbcForeignDataWrapper.java

```diff
@@ -144,9 +146,12 @@ public CompletableFuture<BatchIterator<Row>> getIterator(Role currentUser,
 
         Settings options = server.options();
         String url = urlSetting.get(options);
+        if (!url.startsWith("jdbc:")) {
+            throw new IllegalArgumentException("Invalid JDBC connection string provided: " + url);
```

Comment:
the url might contain credentials, do we really want to log it?

Reply:
oh, good point @proddata !